src/hypofuzz/dashboard.py

```diff
@@ -13,11 +14,13 @@
 import plotly.graph_objects as go
 from dash import dcc, html
 from dash.dependencies import Input, Output
+from hypothesis import settings
 from hypothesis.configuration import storage_directory
 
+from .database import get_db
 from .patching import make_and_save_patches
 
-DATA_TO_PLOT = [{"nodeid": "", "elapsed_time": 0, "ninputs": 0, "branches": 0}]
+DATA_TO_PLOT = []
```

Comment:
We should test that the dashboard doesn't crash if run against an empty database.

Also we should open an issue to have the dashboard show a useful message in that case, to the effect of "The database ({database!r}) doesn't have contain fuzzing reports.  If you've run hypofuzz before, check that the database configured on the workers is the same as for the dashboard."  (after a ~5-10s timeout to avoid the freshly-started-popup problem).

Reply:
Thinking aloud about a future `test_dashboard.py`: define a `dashboard` fixture which starts and returns a `--dashboard-only` fixture. Tests can launch their own `--no-dashboard` process and then use the proposed dashboard-json-api to check for correctness of the fuzzer.

To the problem at hand: how to control the database used by a `--dashboard-only` process? I'd expect with a `--profile` option. Which means `--dashboard-only` needs to execute a pytest collection to pick up on any user-defined profiles? Does this all sound about correct?

(I'm thinking ahead here because the requested database-isolation across tests is difficult given that the hypothesis default is a shared filesystem db).

Reply:
I think `--profile=...` is redundant with `-- --hypothesis-profile=...` (i.e. passing that as a pytest argument); we definitely have to execute a pytest collection step to execute the profile-defining code.

Ideally we'd also have a way to specify a particular database object (so we can use `InMemoryExampleDatabase` for testing); that would need to start a dashboard server within the current process.  Hmm.

Reply:
Good point for `--hypothesis-profile`! Though we may want a convenience alias anyway.

My first thought is to continue running the database via multiprocessing in release, and have a `factory: type[Thread]  | type[Process] = Process` argument which we switch to threading in tests when sharing a db instance. But the api surface isn't quite the same (no `Thread.kill`)..

Reply:
I'm overall -0.5 on the convenience alias; on one hand it _is_ more convenient; but on the other I really want to push people to think about this as "take your pytest invocation and `s/pytest/hypothesis fuzz --/` and you're done"

Reply:
yeah, that's fair.